src/legacy/core_plugins/console/server/elasticsearch_proxy_config.js

```diff
@@ -49,7 +49,8 @@ const createAgent = (legacyConfig) => {
       throw new Error(`Unknown ssl verificationMode: ${verificationMode}`);
   }
 
-  if (legacyConfig.ssl && legacyConfig.ssl.certificateAuthorities.length > 0) {
+  if (legacyConfig.ssl && Array.isArray(legacyConfig.ssl.certificateAuthorities)
+      && legacyConfig.ssl.certificateAuthorities.length > 0) {
```

Comment:
This looks good to me and will fix our issues.

One thing of note is I am curious if certificateAuthorities length will ever be allowed to be zero?

If I change my config to be something like so:

```ts
  ssl:
    verificationMode: 'none'
    certificateAuthorities: []
```

I don't think it gets to this line (which is a good thing) as the validator catches it. But I will still approve this as LGTM :+1:

```ts
- [elasticsearch.ssl.certificateAuthorities.0]: expected value of type [string] but got [Array]
- [elasticsearch.ssl.certificateAuthorities.1]: array size is [0], but cannot be smaller than [1]
    at ObjectType.validate (/Users/frankhassanabad/projects/kibana/packages/kbn-config-schema/target/out/types/type.js:59:19)
    at ConfigService.createConfig (/Users/frankhassanabad/projects/kibana/src/core/server/config/config_service.ts:97:52)
    at MapSubscriber.getDistinctConfig.pipe.operators_1.map.config [as project] (/Users/frankhassanabad/projects/kibana/src/core/server/config/config_service.ts:51:81)
    at MapSubscriber._next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/operators/map.ts:81:29)
    at MapSubscriber.Subscriber.next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/Subscriber.ts:102:12)
    at DistinctUntilChangedSubscriber._next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/operators/distinctUntilChanged.ts:121:24)
    at DistinctUntilChangedSubscriber.Subscriber.next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/Subscriber.ts:102:12)
    at MapSubscriber._next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/operators/map.ts:86:22)
    at MapSubscriber.Subscriber.next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/Subscriber.ts:102:12)
    at MapSubscriber._next (/Users/frankhassanabad/projects/kibana/node_modules/rxjs/src/internal/operators/map.ts:86:22)
  cause:
   { Error: types that failed validation:
       at UnionType.handleError (/Users/frankhassanabad/projects/kibana/packages/kbn-config-schema/target/out/types/union_type.js:36:24)
       at UnionType.onError (/Users/frankhassanabad/projects/kibana/packages/kbn-config-schema/target/out/types/type.js:77:40)
       at internals.Alternatives.Type.schema.error (/Users/frankhassanabad/projects/kibana/packages/kbn-config-schema/target/out/types/type.js:49:53)
       at finish (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/any/index.js:529:50)
       at internals.Alternatives._validate (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/any/index.js:618:24)
       at type._base (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/object/index.js:212:45)
       at type._validate (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/any/index.js:614:37)
       at type._base (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/object/index.js:212:45)
       at type._validate (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/any/index.js:614:37)
       at type._validateWithOptions (/Users/frankhassanabad/projects/kibana/node_modules/joi/lib/types/any/index.js:674:29)
     cause: undefined,
     path: [ 'ssl', 'certificateAuthorities' ],
     errors: [ [SchemaTypeError], [SchemaTypeError] ] } }
server    log   [17:28:23.588] [info][plugins-system] Stopping all plugins.
server    log   [17:28:23.589] [info][plugins][testbed] Stopping plugin
``

Reply:
> One thing of note is I am curious if certificateAuthorities length will ever be allowed to be zero?

You're right, we forbid empty array for `certificateAuthorities`: https://github.com/elastic/kibana/blob/master/src/core/server/elasticsearch/elasticsearch_config.ts#L51-L53. But since `legacyConfig` will be dropped completely soon I tried not to refactor this code without a strong need :)